File: Math/1232.Check If It Is a Straight Line/Solution.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        for i in range(2, len(coordinates)):
            lx, ly = coordinates[i - 2]
            x, y = coordinates[i - 1]
            nx, ny = coordinates[i]
            if (x - lx) * (ny - ly) != (y - ly) * (nx - lx):
                return False
        return True

File: Math/1232.Check If It Is a Straight Line/Solution.java
class Solution {
    public boolean checkStraightLine(int[][] coordinates) {
        for (int i = 2; i < coordinates.length; i++) {
            int lx = coordinates[i - 2][0], ly = coordinates[i - 2][1];
            int x = coordinates[i - 1][0], y = coordinates[i - 1][1];
            int nx = coordinates[i][0], ny = coordinates[i][1];
            if ((x - lx) * (ny - ly) != (y - ly) * (nx - lx)) {
                return false;
            }
        }
        return true;
    }
}